exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -196,3 +194,40 @@ func computeHash(dataArr [][]byte) string {
 	}
 	return fmt.Sprintf("sha256:%x", hash.Sum(nil))
 }
+
+// normalizeData reads content of the resource (configmap or secret) and returns
+// them serialized with constant order. Secret's data is base64 decoded.
+// This is useful to achieve consistent data  between runs, since the content
+// of the data field is a mapp and its order is non deterministic.
+func normalizeData(resource *unstructured.Unstructured) ([][]byte, error) {
+	// Since maps are not ordered, we need to order them to get the same hash at each reconcile.
+	keys := make([]string, 0)
+	data, ok := resource.UnstructuredContent()["data"]
+	if !ok {
+		return nil, errors.New("failed to get data field from the resource")
+	}
+
+	unstructuredData := data.(map[string]interface{})
+	for key := range unstructuredData {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	dataList := make([][]byte, 0)
+	for _, key := range keys {
+		val, ok, err := unstructured.NestedString(unstructuredData, key)
+		if !ok || err != nil {
+			return nil, errors.New("failed to get value field from the resource")
```

Comment:
```suggestion
			return nil, errors.Wrapf(err, "failed to get value field from the resource")
```
nit: What about this way?

Reply:
I separated err and !ok cases and added more info to the error, let me know

Reply:
Looks good, thx!